programs/util.c

```diff
@@ -87,6 +94,239 @@ U32 UTIL_isDirectory(const char* infilename)
     return 0;
 }
 
+int UTIL_createDir(const char* outDirName)
+{
+    int r;
+    if (UTIL_isDirectory(outDirName))
+        return 0;   /* no need to create if directory already exists */
+
+#if defined(_MSC_VER) || defined(__MINGW32__) || defined (__MSVCRT__)
+    r = _mkdir(outDirName);
+    if (r || !UTIL_isDirectory(outDirName)) return 1;
+#else
+    r = mkdir(outDirName, S_IRWXU | S_IRWXG | S_IRWXO); /* dir has all permissions */
+    if (r || !UTIL_isDirectory(outDirName)) {
+        errno = 0;
+        return 1;
+    }
+#endif
+    errno = 0;
+    return 0;   /* success */
+}
+
+/* function requires posix >= 2001 */
+int UTIL_getRealPath(const char* relativePath, char* absolutePath) {
+#if defined(_MSC_VER) || defined (__MINGW32__) || defined (__MSVCRT__)
+    if (_fullpath(absolutePath, relativePath, LIST_SIZE_INCREASE) == NULL)
+        return 1;
+#elif (PLATFORM_POSIX_VERSION >= 200112L)
+#define _BSD_SOURCE
+    char* r;
+    r = realpath(relativePath, absolutePath);
+    if (r == NULL && errno != ENOENT /* we will be using this to generate "FAKE" paths, so ignore this err */) {
```

Comment:
Does this work? Do you get the results that you expect if you try to realpath something that doesn't exist? My read of the man page suggests that you won't.

Reply:
Should be fixed in the latest version.